src/javascript/Switch.js

```diff
@@ -39,6 +39,10 @@ export class Switch {
             this.isMaterial = true;
         }
 
+        if (this.element.disabled) {
+            this.disabled = true;
+        }
```

Comment:
Keeping this in sync is likely to be a little painful. I'm guessing you're doing this to play better with some framework, and in which case I would say that it might be better to use getters and setters here, ie something like:

```javascript
get disabled() {
  return this.element.disabled;
}

// newVal should be a boolean
set disabled(newVal) {
  this.track.classList.add("disabled");
}
```

Alternatively, if this pattern works better, consider refactoring this to:

```javascript
this.disabled = this.element.disabled;
```

That way `this.disabled` always has a default value.

Reply:
I can definitely do the latter option here. I'm not sure why I didn't do that to begin with. I don't really need setters or getters. My intent is to have this simply read the `disabled` attribute from the checkbox and use that to determine whether to disable the switch.

Reply:
That's partially why I suggested the getters and setters&mdash;they may be overkill for your purposes, but using them to basically pipe state back and forth between the DOM and the Javascript makes it easier to keep everything in sync by making sure you're not also managing your own `disabled` boolean in memory, but instead just binding `switchInstance.disabled` to the DOM's state.

Reply:
That makes sense. I haven't really used getters and setters much. When will `set disabled` be called in your example? Will that somehow be called automatically when the checkbox's `disabled` attribute changes?

Reply:
> When will `set disabled` be called in your example?

Whenever the `disabled` property on the switch object is set, ie. presuming you have an instance of the simple switch called `instance`, it would be set when some caller does:

```javascript
instance.disabled = true;
// or
instance.disabled = false;
// it also doesn't care about input type, you need to manually validate this if desired
instance.disabled = "foobar";
```

>Will that somehow be called automatically when the checkbox's `disabled` attribute changes?

Probably not, but if this change is for a specific framework, the framework _might_ be smart enough to understand the getter and poll `.disabled` from time to time.  Otherwise, you would need to use a [MutationObserver](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver) to watch the element's attributes / state and then reflect that upwards appropriately for your framework (whether that means setting `this.disabled` or emitting an event or some other thing).

Reply:
Got it. That's what I thought. 🤔 I wonder if it would make sense to have that MutationObserver as part of this library. How would you feel about that? I view this library as a wrapper over a checkbox to reskin it as a switch, so having the user manually update the `disabled` prop feels counter to that.

Reply:
I also view this as a wrapper over a checkbox, so the closer that this stays to matching checkbox functionality in form and function the better IMO, and to that end I think I agree: it 100% makes sense to have the MutationObserver-bits just hidden within this library.